From 7f4fde2eea94e2353be52eef478304ca728132bd Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Tom=C3=A1=C5=A1=20Ku=C4=8Da?= <tomas.kuca@matfyz.cz>
Date: Wed, 19 Aug 2009 03:03:43 +0200
Subject: [PATCH] Cache format fix for XMLHttpRequests

* cache_format now returns the same format as template_format
---
 actionpack/lib/action_dispatch/http/request.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_dispatch/http/request.rb b/actionpack/lib/action_dispatch/http/request.rb
index bff030f..bb860c3 100755
--- a/actionpack/lib/action_dispatch/http/request.rb
+++ b/actionpack/lib/action_dispatch/http/request.rb
@@ -205,7 +205,7 @@ module ActionDispatch
     end
 
     def cache_format
-      parameters[:format]
+      template_format.to_s
     end
 
     # Returns true if the request's "X-Requested-With" header contains
-- 
1.6.0.4

